jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/MethodScope.java

```diff
@@ -171,11 +177,49 @@ public boolean isGetter() {
     public <A extends Annotation> A getAnnotation(Class<A> annotationClass) {
         A annotation = super.getAnnotation(annotationClass);
         if (annotation == null) {
-            annotation = this.getRawMember().getAnnotatedReturnType().getAnnotation(annotationClass);
+            annotation = findGetterCandidates(this.getRawMember()).stream().map(m -> {
+                return Optional.ofNullable(m.getAnnotation(annotationClass))
+                        .orElse(m.getAnnotatedReturnType().getAnnotation(annotationClass));
+            }).filter(Objects::nonNull).findFirst().orElse(null);
         }
         return annotation;
     }
 
+    private Collection<Method> findGetterCandidates(Method m) {
+        List<Method> candidates = new ArrayList<>();
+        addGetterCandidates(candidates, m.getDeclaringClass(), m.getName());
+        return candidates;
+    }
+
+    private void addGetterCandidates(List<Method> candidates, Class<?> declaringClass,
```

Comment:
**suggestion (non-blocking)**: If `addGetterCandidates()List<Method>, Class<?>, String)` were not `void` but returned the possibly extended`List<Method>`, this code could be shortened a bit.
```suggestion
        return addGetterCandidates(new ArrayList<>(), m.getDeclaringClass(), m.getName());
    }

    private List<Method> addGetterCandidates(List<Method> candidates, Class<?> declaringClass,
```
Might as well remove `findGetterCandidates(Method)`/`findGetterCandidates()` then.